src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,110 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String USER_PASSWORD = "54321paSs";
+    private static final int USER_AGE = 21;
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private final StorageDao storageDao = new StorageDaoImpl();
+
+    @Test
+    public void register_userWithSameLoginAlreadyExist_notOk() {
+        String userLogin = "Ali852";
+        User firstUser = createUser(userLogin, USER_PASSWORD, USER_AGE);
+        storageDao.add(firstUser);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(firstUser);
+        });
+    }
+
+    @Test
+    public void register_loginIsNull_notOk() {
+        User userWithLoginIsNull = createUser(null, USER_PASSWORD, USER_AGE);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(userWithLoginIsNull);
+        });
+    }
```

Comment:
I would additionally check whether the login is not empty, doesn't start with symbols, numbers,  doesn't have any uppercase letters, and contains "@" symbol

Reply:
I understand your point, but I don't see the need for it, since the task does not specify a check for a valid email like login.
I will wait for the mentor's comment ;)